torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -1613,4 +1643,93 @@ PyTorchModelLoader::PyTorchModelLoader(
   error = llvm::Error::success();
 }
 
+/// Takes a glow::Function \p F, a jit::Graph \p subgraph to load, detects
+/// inputs for the subgraph automatically. \returns error on failure.
+/*static*/
+llvm::Error PyTorchModelLoader::parseJITGraph(
+    glow::Function &F, const torch::jit::Graph &graph,
+    const at::ArrayRef<torch::jit::IValue> &inputs,
+    const at::ArrayRef<at::Tensor> &parameters,
+    std::vector<glow::Placeholder *> &inputPlaceholders,
+    std::vector<glow::Placeholder *> &outputPlaceholders) {
+  llvm::Error error = llvm::Error::success();
+  MARK_ERR_CHECKED(error);
+  PyTorchModelLoader loader(F, graph, inputs, parameters, inputPlaceholders,
+                            outputPlaceholders, error);
+  return error;
+}
+
+PyTorchModelLoader::PyTorchModelLoader(
+    glow::Function &F, const torch::jit::Graph &graph,
+    const at::ArrayRef<torch::jit::IValue> &inputs,
+    const at::ArrayRef<at::Tensor> &parameters,
+    std::vector<glow::Placeholder *> &inputPlaceholders,
+    std::vector<glow::Placeholder *> &outputPlaceholders, llvm::Error &error)
+    : F_(F), inputs_(inputs), copyTensorMemory_(true) {
+  auto graphInputValues = graph.inputs();
+  if (inputs.size() + parameters.size() != graphInputValues.size()) {
+    error = MAKE_ERR(glow::strFormat(
+        "Number of Graph inputs %lu must match the "
+        "number of placeholders %lu + number of "
+        "provided inputs %lu.",
+        graphInputValues.size(), parameters.size(), inputs.size()));
+    return;
+  }
+
+  size_t graphIdx = 0;
+  for (size_t i = 0; i < inputs.size(); ++i, ++graphIdx) {
+    const torch::jit::Value *inputValue = graphInputValues[graphIdx];
+    const c10::IValue inputIValue = inputs.at(i);
+    if (inputIValue.isTensor()) {
+      const auto ptTensorType = c10::TensorType::create(inputIValue.toTensor());
+      glow::Placeholder *PH;
+      if (auto resOrErr = loadValue(inputValue, *ptTensorType)) {
+        PH = std::move(*resOrErr);
+      } else {
+        error = resOrErr.takeError();
+        return;
+      }
+      inputPlaceholders.push_back(PH);
+    } else {
+      error =
+          MAKE_ERR(glow::strFormat("Only tensor type is supported for now."));
+      return;
+    }
+  }
+
+  for (size_t i = 0; i < parameters.size(); ++i, ++graphIdx) {
```

Comment:
We shouldn't have to duplicate this constructor, we could just have parameters be empty by default and everything else is exactly the same right?

Reply:
there is no settings as well.

Reply:
Refactored, I think having two constructors is ok, they have different parameters and do different things with some overlapped (I have moved common code into the separate methods).